### ISSUE TITLE
Add more loss functions like cross-entropy...

### ISSUE THREAD

**Issue by TarunTomar122** [RoboticsClubIITJ/ML-DL-implementation#9]
Right we have implemented only Mean Squared Error and Logarithmic Error as our loss functions in our module. As we are going to implement more and more machine learning algorithms we will need more loss functions. There in this issue you can add one or more loss functions like cross-entropy etc.

**Comment by Saptashrungi** [comment]
I would like to work on this issue. I will share the loss functions I would be working on by tomorrow.

**Comment by quadri-haider-ali** [comment]
I would love to work on this part.

**Comment by rohansingh9001** [comment]
@quadri-haider-ali Sure go ahead. Issue assigned to you.

**Comment by SaiSrichandra** [comment]
I would also like to work on this issue

**Comment by TarunTomar122** [comment]
@SaiSrichandra  You can work on this... Do Tell prior what loss function are you going to add?

**Comment by SaiSrichandra** [comment]
I will add the KL Divergence loss

**Comment by TarunTomar122** [comment]
Okay @SaiSrichandra do keep us updated

**Comment by parva-jain** [comment]
Hey I would like to add mean squared logarithmic error(msle) and root mean squared logarithmic error(rmlse).

**Comment by rohansingh9001** [comment]
@parva-jain sure go ahead. Keep us updated on your progress and feel free to ask anything in the Gitter channel for further clarification.

**Comment by Halix267** [comment]
Hey @rohansingh9001  I would like to add root mean squared logarithmic error (rmsle)

**Comment by parva-jain** [comment]
> Hey @rohansingh9001 I would like to add root mean squared logarithmic error (rmsle)

Actually, I'm ready with the loss function of mean squared logarithmic error and derivative equation of mean squared logarithmic error(handly written) and only left with its implementation in python. Also @rohansingh9001, the equation is a bit complex so how should I justify it?

**Comment by Vinit-source** [comment]
Hey I would like to add the Triplet loss.
Refer [this](https://en.wikipedia.org/wiki/Triplet_loss#:~:text=Triplet%20loss%20is%20a%20loss,a%20negative%20(falsy)%20input.) for more info. on it.

**Comment by Udit-git-acc** [comment]
I would like to add cosine similarity loss function.

**Comment by kwanit1142** [comment]
@Vinit-source and @Udit-git-acc, Okay 👍

**Comment by Siddharth-Singh27** [comment]
I would like to add logcosh_loss function

**Comment by kwanit1142** [comment]
@Siddharth-Singh27 sure, go on

**Comment by kwanit1142** [comment]
Completion Phase-1

Huber Loss -> Shreya Sachan
Cosine Similarity -> Udit Agarwal
Mean Squared log loss -> Devyani
Log cosh loss -> Siddharth Singh

**Comment by dr-ghost** [comment]
I would like to add the Binary Crossentropy loss function

**Comment by kwanit1142** [comment]
Sure, go ahead

**Comment by testgithubtiwari** [comment]
I want to add hinge function loss , Will I ?

**Comment by kwanit1142** [comment]
Sure, Go ahead :) Just check in the files, whether hinge function loss is already there or not?

**Comment by yorozuya-2003** [comment]
I would like to add the Poisson Loss function.

**Comment by kwanit1142** [comment]
Sure, Go ahead :) Just check in the files, whether Poisson function loss is already there or not?
